app/controllers/api_controller/service_templates.rb

```diff
@@ -114,13 +117,19 @@ def refresh_dialog_fields_service_template(_object, st, data = {})
       action_result(false, err)
     end
 
-    def define_service_template_dialog(st, dialog_fields)
-      resource_action = st.resource_actions.find_by_action("Provision")
-      workflow = ResourceActionWorkflow.new({}, @auth_user_obj, resource_action, :target => st)
+    def define_service_template_dialog(st, ra, dialog_fields)
+      workflow = ResourceActionWorkflow.new({}, @auth_user_obj, ra, :target => st)
       dialog_fields.each { |key, value| workflow.set_value(key, value) }
       workflow.dialog
     end
 
+    def fetch_resource_action(st, data)
+      data ||= {}
+      return st.resource_actions.find_by_id(data["id"]) if data["id"].present?
```

Comment:
@abellotti Unfortunately, this won't work, because the resource action isn't attached to the service template that is coming in to the request, it's attached to a different service template. It's kind of weird because we don't really care about the service template or catalog in this case, but all we need is a workflow built from the specific resource action.